language/bytecode-verifier/src/signature.rs

```diff
@@ -241,6 +309,15 @@ impl<'a> SignatureChecker<'a> {
         constraints: &[AbilitySet],
         global_abilities: &[AbilitySet],
     ) -> PartialVMResult<()> {
+        if type_arguments.len() != constraints.len() {
```

Comment:
Question: I don't seem to find where this check was performed... Did I miss anything in the bounds check or in the signature check?